src/dialogs/more-info/controls/more-info-camera.ts

```diff
@@ -34,14 +37,52 @@ class MoreInfoCamera extends LitElement {
         allow-exoplayer
         controls
       ></ha-camera-stream>
+
+      <div class="actions">
+        <ha-progress-button
+          @click=${this._downloadSnapshot}
+          .disabled=${this.stateObj.state === UNAVAILABLE}
+        >
+          ${this.hass.localize(
+            "ui.dialogs.more_info_control.camera.download_snapshot"
+          )}
+        </ha-progress-button>
+      </div>
     `;
   }
 
+  private async _downloadSnapshot() {
+    const result: Response | undefined = await this.hass?.callApiRaw(
+      "GET",
+      `camera_proxy/${this.stateObj!.entity_id}`
+    );
+
+    if (!result) {
+      return;
+    }
+
+    const blob = await result.blob();
+    const url = window.URL.createObjectURL(blob);
+    fileDownload(url);
```

Comment:
We should have some error handling here and a progress indicator in case of a slow network. You already use `ha-progress-button` so you can access it from the even like this `const button = ev.currentTarget as any;`. Have a look at how this is done in other components